src/main/java/com/fakng/fakngagrgtr/parser/stripe/StripeParser.java

```diff
@@ -0,0 +1,150 @@
+                        @Value("${url.stripe}") String url) {
+        super(htmlWebClient, companyRepository, locationProcessor);
+        this.url = url;
+    }
+
+    @PostConstruct
+    public void init() {
+        initBase();
+    }
+
+    @Override
+    protected String getCompanyName() {
+        return "Stripe";
+    }
+    @Override
+    public List<Vacancy> getAllVacancies() throws IOException {
+        HtmlPage page = downloadPage(String.format(url, 0));
+        int pageCount = getPageCount(page);
+        List<Vacancy> vacancyList = new ArrayList<>(fillVacancy(
+                page.getByXPath(VACANCIES_URLS_XPATH), page.getByXPath(VACANCY_CITIES_XPATH)
+        ));
+
+        for (int i = 1; i <= pageCount; i++) {
+            page = downloadPage(String.format(url, i));
+            vacancyList.addAll(fillVacancy(
+                    page.getByXPath(VACANCIES_URLS_XPATH), page.getByXPath(VACANCY_CITIES_XPATH)
+            ));
+        }
+
+        return vacancyList;
+    }
+
+    private int getPageCount(HtmlPage page) {
+        String url = ((DomElement) page.getByXPath(JOBS_PAGINATION_LIST_XPATH).get(0)).getAttribute("href");
+        return Integer.parseInt(url.replaceAll("\\D+", "")) / PAGE_SIZE;
+    }
+
+    private List<Vacancy> fillVacancy(List<DomNode> urls, List<DomElement> cities) throws IOException {
+        List<Vacancy> vacancies = new ArrayList<>();
+
+        for (int i = 0; i < urls.size(); i++) {
+            String jobId = urls.get(i).getAttributes().getNamedItem("href").getNodeValue();
+
+            // Skip repeating vacancies and parse locations for them
+            List<LocationDTO> locationDTOs = new ArrayList<>();
+            while (jobId.equals(urls.get(i).getAttributes().getNamedItem("href").getNodeValue())) {
+                locationDTOs.add(createLocationDto(cities.get(i)));
+                i++;
+                if (i >= urls.size()) break;
+            }
+
+            // Variable i stopped in incorrect position
+            i--;
+            vacancies.add(createVacancy(jobId, locationDTOs));
+        }
+
+        return vacancies;
+    }
+
+    private Vacancy createVacancy(String jobUrl, List<LocationDTO> locationDTOS) throws IOException {
+        Vacancy vacancy = new Vacancy();
+
+        HtmlPage page = downloadPage(STRIPE_URL + jobUrl);
+        DomElement title = (DomElement) page.getByXPath(VACANCY_TITLE_XPATH).get(0);
+        DomElement jobType = ((DomElement) page.getByXPath(VACANCY_JOB_TYPE_XPATH).get(0)).getNextElementSibling();
+        DomElement team = ((DomElement) page.getByXPath(VACANCY_TEAM_XPATH).get(0)).getNextElementSibling();
+
+        vacancy.setTitle(title.getTextContent());
+        vacancy.setDescription(createFullDescription(
+                createDescription(page), jobType.getTextContent(), team.getTextContent())
+        );
+        vacancy.setUrl(STRIPE_URL + jobUrl);
+        vacancy.setJobId(jobUrl.replaceAll("\\D+", ""));
```

Comment:
What will happen, if there will be a digit in a vacancy title?

What you are probably looking for is the last numeric sequence in the URL. Can't you find the last appearance of /? Everything after it will be your jobId, I guess. Besides, regex is pretty slow thing

Reply:
Can I use that /.*/ regex expression or I need to get job id without regex at all?